### PR TITLE
Some small refactorings

### PR DESCRIPTION
**Pull request by alex1701c**

### CODE REVIEW

**Comment by alex1701c** [comment]
@dragotin I saw a post in planet.kde.org and gave the source code a quick look :) 

The unused params in the last commit could be entirely removed:
```diff
diff --git a/xrview/xrechnung.cpp b/xrview/xrechnung.cpp
index 11d9db6..15cbcbb 100644
--- a/xrview/xrechnung.cpp
+++ b/xrview/xrechnung.cpp
@@ -101,7 +101,7 @@ void XRechnung::slotErrorOccurred(QProcess::ProcessError error)
     }
 }
 
-void XRechnung::slotDomFinished(int /*exitCode*/, QProcess::ExitStatus /*exitStatus*/)
+void XRechnung::slotDomFinished()
 {
     QProcess *p = qobject_cast<QProcess*>(sender());
 
@@ -134,7 +134,7 @@ void XRechnung::createHtml()
     process->start("java", args);
 }
 
-void XRechnung::slotHtmlFinished(int /*exitCode*/, QProcess::ExitStatus /*exitStatus*/)
+void XRechnung::slotHtmlFinished()
 {
     QProcess *p = qobject_cast<QProcess*>(sender());
 
diff --git a/xrview/xrechnung.h b/xrview/xrechnung.h
index 7599607..f71272c 100644
--- a/xrview/xrechnung.h
+++ b/xrview/xrechnung.h
@@ -47,9 +47,9 @@ private slots:
     bool readDomDocument();
     void createUbl();
     void slotErrorOccurred(QProcess::ProcessError error);
-    void slotDomFinished(int exitCode, QProcess::ExitStatus exitStatus);
+    void slotDomFinished();
     void createHtml();
-    void slotHtmlFinished(int exitCode, QProcess::ExitStatus exitStatus);
+    void slotHtmlFinished();
 
 private:
     QUrl         _url;
```

Though I am not sure if you'd like that.

I only tested the compilation & startup of the app, because I do not have a xrechnung file at hand. But the changes should only be relevant at compile time.

File: xrview/xrechnung.cpp [hunk 1]
```diff
@@ -74,19 +74,19 @@ QDomDocument XRechnung::domDocument() const
...
+        SaxonJar,
+        QString("-s:%1").arg(file),
+        QString("-xsl:%1").arg(XslUBL),
+        QString("-o:%1").arg(ublFileName(_url)),
+    };
 
     QProcess *process = new QProcess;
 
-    connect(process, QOverload<int, QProcess::ExitStatus>::of(&QProcess::finished),
-                this, &XRechnung::slotDomFinished);
```
Comment (dragotin) [inline xrview/xrechnung.cpp:95]: This is to keep Qt5 compatibility, see https://stackoverflow.com/questions/12534367/qt-no-such-slot-for-the-qprocessfinished-signal that describes the problem. Are you sure your suggested change works with Qt5?

File: xrview/xrechnung.cpp [hunk 2]
```diff
@@ -101,11 +101,8 @@ void XRechnung::slotErrorOccurred(QProcess::ProcessError error)
     }
 }
 
-void XRechnung::slotDomFinished(int exitCode, QProcess::ExitStatus exitStatus)
+void XRechnung::slotDomFinished(int /*exitCode*/, QProcess::ExitStatus /*exitStatus*/)
 {
-    Q_UNUSED(exitStatus)
-    Q_UNUSED(exitCode)
```
Comment (dragotin) [inline xrview/xrechnung.cpp:113]: I'd rather implement a check and at least give a debug message if the call failed.

**Review by dragotin — CHANGES_REQUESTED** [review]
Very nice, thanks, but a few comments.

File: xrview/xrechnung.cpp (hunk 1 above)
Comment (alex1701c) [reply xrview/xrechnung.cpp:95]: Yes, when disabling deprecated public API the other signal is no longer visible and thus it is not overleaded.
(I have compiled this project with Qt5)

This is what allowed me to clean up lots of qOverloads in KDE :)

File: xrview/xrechnung.cpp (hunk 2 above)
Comment (alex1701c) [reply xrview/xrechnung.cpp:113]: The standard error is printed out unconditionally. Also, we are connected to the errorOccurred signal. 

Should we maybe print out other errors than `FailedToStart` in slotErrorOccurred?

**Pull request merged**